boot/bootstate20.go

```diff
@@ -341,11 +342,8 @@ func (ks20 *bootState20Kernel) setNext(next snap.PlaceInfo, bootCtx NextBootCont
 		}
 		// kernels are usually loaded directly by the bootloader, for
 		// which we may need to pass additional data to make 'try'
-		// operation more robust - that might be provided by the
-		// RebootBootloader interface
-		if rbi.RebootBootloader, err = ks20.getRebootBootloader(); err != nil {
```

Comment:
should getRebootBootloader go away? should we store the options only if it's a RebootBootloader?

Reply:
yea good point, the `getRebootBootloader` should go away. I think it's fine if getRebootArguments handles this check if it actually implements the right interface